driver/razermouse_driver.c

```diff
@@ -443,6 +443,23 @@ static ssize_t razer_attr_write_mode_static(struct device *dev, struct device_at
             report.transaction_id.id = 0xff;
             break;
 
+        case USB_DEVICE_ID_RAZER_NAGA_TRINITY:
+            ;
+            // Some sort of mode switcher required after initialization and before color switching
+            report = get_razer_report(0x0f, 0x02, 0x06);
+            report.arguments[0] = 0x00;
+            report.arguments[1] = 0x00;
+            report.arguments[2] = 0x08;
+            report.arguments[3] = 0x00;
+            report.arguments[4] = 0x00;
+            report.arguments[5] = 0x00;
+            report.transaction_id.id = 0x1f;
+
+            razer_send_payload(usb_dev, &report);
```

Comment:
I'm not happy about this section, but I'm not sure where to move it if it's really required :(

Reply:
Yea, it's completely different than other payloads that have established methods. I wasn't sure where to put it either.